tests/tests/test_os_ent_migration.py

```diff
@@ -0,0 +1,240 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import pytest
+import logging
+import subprocess
+import requests
+import time
+import json
+
+from ..conftest import docker_compose_instance
+
+from ..common_docker import *
+from .conductor import Conductor
+
+from testutils.common import create_user, make_accepted_device, Device, User, Tenant
+from testutils.api.client import ApiClient
+from testutils.infra.cli import CliTenantadm
+import testutils.api.deviceauth as deviceauth_v1
+import testutils.api.deviceauth_v2 as deviceauth_v2
+import testutils.api.deployments as deployments
+import testutils.api.useradm as useradm
+
+@pytest.fixture(scope="class")
+def initial_os_setup():
+    """ Start the minimum OS setup, create some uses and devices. 
+        Return {"os_devs": [...], "os_users": [...]}
+    """
+
+    stop_docker_compose()
+
+    docker_compose_cmd("-f " + COMPOSE_FILES_PATH + "/docker-compose.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.testing.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.storage.minio.yml up -d",
+                       use_common_files=False)
+    ensure_conductor_ready(60, 'provision_device')
+
+    init_data = initialize_os_setup()
+
+    return init_data
+
+@pytest.fixture(scope="class")
+def initial_enterprise_setup(initial_os_setup):
+    """
+        Start ENT for the first time (no tenant yet).
+    """
+    stop_docker_compose_exclude(['mender-mongo'])
+
+    docker_compose_cmd("-f " + COMPOSE_FILES_PATH + "/docker-compose.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.enterprise.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.testing.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.storage.minio.yml up -d",
+                       use_common_files=False)
+
+    return initial_os_setup
+
+@pytest.fixture(scope="class")
+def migrated_enterprise_setup(initial_enterprise_setup):
+    """
+        Create an org (tenant + user), restart with default tenant token.
+        The ENT setup is ready for tests.
+        Return {"os_devs": [...], "os_users": [...], "tenant": <Tenant>}
+    """
+    ent_data = migrate_ent_setup()
+
+    # preserve the user/tenant created before restart
+    stop_docker_compose_exclude(['mender-mongo'])
+
+    docker_compose_cmd("-f " + COMPOSE_FILES_PATH + "/docker-compose.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.enterprise.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.testing.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.testing.enterprise.yml \
+                        -f " + COMPOSE_FILES_PATH + "/docker-compose.storage.minio.yml up -d --no-recreate",
```

Comment:
Does `--no-recreate` matter? Isn't all persistent data in the volumes? If it does, this needs to be included in the production documentation as well.

Reply:
it matters in tests only. it was just easier to preserve data via no recreate, than fiddling with env in dockerfiles.

if a volume persists in prod install - that's an equivalent.

Reply:
Ok, thanks for confirming.